lib/rules/prefer-screen-queries.ts

```diff
@@ -113,52 +136,54 @@ export default createTestingLibraryRule<Options, MessageIds>({
           // save the destructured query methods as safe since they are coming
           // from within or render + base/container options
           saveSafeDestructuredQueries(node);
-          return;
-        }
-
-        if (ASTUtils.isIdentifier(node.id)) {
+        } else if (ASTUtils.isIdentifier(node.id)) {
           withinDeclaredVariables.push(node.id.name);
         }
       },
-      'CallExpression > Identifier'(node: TSESTree.Identifier) {
-        if (!helpers.isBuiltInQuery(node)) {
+      CallExpression(node) {
+        const identifierNode = getDeepestIdentifierNode(node);
+
+        if (!identifierNode) {
           return;
         }
 
-        if (
-          !safeDestructuredQueries.some((queryName) => queryName === node.name)
-        ) {
-          reportInvalidUsage(node);
+        if (helpers.isRenderUtil(identifierNode)) {
+          detectRenderWrapper(identifierNode);
         }
-      },
-      'MemberExpression > Identifier'(node: TSESTree.Identifier) {
-        function isIdentifierAllowed(name: string) {
-          return ['screen', ...withinDeclaredVariables].includes(name);
+
+        if (!helpers.isBuiltInQuery(identifierNode)) {
+          return;
         }
 
-        if (!helpers.isBuiltInQuery(node)) {
+        if (!isMemberExpression(identifierNode.parent)) {
+          const isSafeDestructuredQuery = safeDestructuredQueries.some(
+            (queryName) => queryName === identifierNode.name
+          );
+          if (isSafeDestructuredQuery) {
+            return;
+          }
+
+          reportInvalidUsage(identifierNode);
```

Comment:
Is there a reason why you're not returning this function call?

It's `void`, so doing `return;` or returning a function with a `void` return type is the same, no?

Reply:
Because of [`consistent-return` rule](https://eslint.org/docs/rules/consistent-return)
<img width="613" alt="CleanShot 2021-05-22 at 22 50 38@2x" src="https://user-images.githubusercontent.com/2677072/119240505-2fe13c00-bb50-11eb-86df-9142342dbccf.png">